lms/djangoapps/certificates/badge_handler.py

```diff
@@ -0,0 +1,151 @@
+"""
+BadgeHandler object-- used to award Badges to users who have completed courses.
+"""
+import hashlib
+import logging
+import mimetypes
+import requests
+
+from django.conf import settings
+from django.contrib.sites.models import Site
+from django.core.urlresolvers import reverse
+from eventtracking import tracker
+from lazy import lazy
+from requests.packages.urllib3.exceptions import HTTPError
+from certificates.models import BadgeAssertion
+from student.models import CourseEnrollment
+from xmodule.modulestore.django import modulestore
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BadgeHandler(object):
+    """
+    The only properly public method of this class is 'award'. If an alternative object is created for a different
+    badging service, the other methods don't need to be reproduced.
+    """
+    # Global caching dict
+    badges = {}
+
+    def __init__(self, course_key):
+        self.course_key = course_key
+        assert settings.BADGR_API_TOKEN
+
+    @lazy
+    def base_url(self):
+        """
+        Base URL for all API requests.
+        """
+        return "{}/v1/issuer/issuers/{}".format(settings.BADGR_BASE_URL, settings.BADGR_ISSUER_SLUG)
```

Comment:
@antoviaque @kelketek I haven't had a chance to look at all this code, but if you want to keep an eye toward issuing badges in overlapping contexts, I think it might be better to be more flexible  in this setting. I'd eventually hope edX would be interested in using API calls outside of `../issuer/`. Cheers!